app/models/library.rb

```diff
@@ -165,6 +164,18 @@ def file_last_modified(file)
     end
   end
 
+  def create_path_if_not_on_disk=(val)
+    if val == "1" && storage_service == "filesystem"
+      begin
+        FileUtils.makedirs(path)
+      rescue Errno::EROFS
+        errors.add(:path, "must be writable")
```

Comment:
if you change the string to `:non_writable`, it will use the translations properly. The key already exists from the path validator, so that should be all you need to do.